app/services/user_alerts/alert_user_about_new_device.rb

```diff
@@ -2,17 +2,17 @@
 
 module UserAlerts
   class AlertUserAboutNewDevice
-    def self.call(user, device, disavowal_token)
+    def self.call(event:, device:, disavowal_token:)
       if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
-        user.sign_in_new_device_at ||= Time.zone.now
-        user.save
+        event.user.sign_in_new_device_at ||= event.created_at
+        event.user.save
```

Comment:
is the workflow different here than in
https://github.com/18F/identity-idp/pull/10370/files#diff-e3f99660c095fef2341988d158c40d0a47ab706abf98070b32cb4a7ce4b02b71R19-R21

Reply:
Yeah, the other code path accounts for a user signing in using their PIV ("Sign in with your government employee ID" on the Sign In page). In that flow, they never submit their email and password, so we don't get that `sign_in_new_device_at` assigned.